### PR TITLE
syslinux: fix submodule clone

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/syslinux/default.nix [hunk 1]
```diff
@@ -68,12 +79,16 @@ stdenv.mkDerivation {
...
-    substituteInPlace utils/ppmtolss16 --replace /usr/bin/perl $(type -P perl)
+    substituteInPlace Makefile --replace-fail /bin/pwd $(type -P pwd)
+    substituteInPlace utils/ppmtolss16 --replace-fail /usr/bin/perl $(type -P perl)
 
     # fix tests
     substituteInPlace tests/unittest/include/unittest/unittest.h \
-      --replace /usr/include/ ""
+      --replace-fail /usr/include/ ""
+
+    # needs to be manually put in place because submodule hash contains an extra g in the beginning gd34132e
```
Comment (raboof) [inline pkgs/os-specific/linux/syslinux/default.nix:89]: I'm not sure the `g` is the problem here: the error message in the current behavior has `fatal: Fetched in submodule path 'gnu-efi', but it did not contain d34132e62f666904158c7ec2f1eef5a9d5281c36. Direct fetching of that commit failed.`, so it seems to get the git hash just fine.

**Review by raboof — APPROVED** [review]
Tested that it works in a `nix-shell` (with `unpackPhase` and `patchPhase`). Not sure I understand the problem, but being able to manipulate gnu-efi-src might be nice anyway, so LGTM.

File: pkgs/os-specific/linux/syslinux/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/os-specific/linux/syslinux/default.nix:89]: Then my best next guess is, that sourceforge is weird. I didn't want to click through many pages to find the old release.

**Comment by samueldr** [comment]
Is there a linked issue? What problem does that solve?

I don't see a rationale anywhere about this, or a concise way to reproduce an issue.

File: pkgs/os-specific/linux/syslinux/default.nix [hunk 2]
```diff
@@ -10,17 +11,27 @@
 , python3
 }:
 
+let
+  gnu-efi-src = fetchFromGitHub {
+    owner = "ncroxon";
+    repo = "gnu-efi";
+    # see 363d61c4f112b972649b19d67e96b9321f738f00 in syslinux
```
Comment (samueldr) [inline pkgs/os-specific/linux/syslinux/default.nix:18]:
```suggestion
    # see 363d61c4f112b972649b19d67e96b9321f738f00 in syslinux
    # https://repo.or.cz/syslinux.git/commitdiff/363d61c4f112b972649b19d67e96b9321f738f00
```

**Comment by samueldr** [comment]
I took some time into actually re-familiarizing myself with the derivation, then fixed the build so we use `gnu-efi` from Nixpkgs to build the EFI outputs. See #317030.

**Pull request closed without merging**